caffe2/python/rnn_cell.py

```diff
@@ -481,12 +478,10 @@ def _prepare_output_sequence(self, model, state_outputs):
     def _apply_dropout(self, model, output):
         if self.dropout_ratio and not self.forward_only:
             with core.NameScope(self.name or ''):
-                output, _ = model.net.Dropout(
+                output = brew.dropout(
+                    model,
                     output,
-                    [
-                        str(output) + '_with_dropout_mask{}'.format(self.mask),
-                        str(output) + '_dropout_mask{}'.format(self.mask),
-                    ],
+                    str(output),
```

Comment:
Here was unfortunately a bug gone undetected. The output should not be in-place, that causes problems with gradient generation in some cases.

Reply:
Fix is submitted, landing soon